lib/datadog/tracing/contrib/karafka/monitor.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require_relative 'ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Karafka
+        # Custom monitor for Karafka
+        class Monitor < ::Karafka::Instrumentation::Monitor
+          TRACEABLE_EVENTS = %w[
+            worker.processed
```

Comment:
Could we subscribe to the event `worker.processed` (e.g. `Karafka.monitor.subscribe 'worker.processed' do |event|`), instead of creating a custom `::Karafka::Instrumentation::Monitor`?

I ask because in `Patcher::patch` we have `::Karafka::App.config.monitor = Monitor.new`, and I'm thinking that this overrides whatever other monitors the application already had (e.g. Sentry error monitoring). I'm wondering if `Karafka.monitor.subscribe 'worker.processed'` would be less destructive.

Reply:
you can do that yes but you wont have ability to `yield` around to instrument with wrapping.